datadog_checks_base/datadog_checks/base/utils/db/core.py

```diff
@@ -0,0 +1,124 @@
+            and 'global_custom_queries' in self.check.init_config
+            and is_affirmative(use_global_custom_queries)
+        ):
+            custom_queries = self.check.init_config.get('global_custom_queries', [])
+
+        # Deduplicate
+        for i, custom_query in enumerate(iter_unique(custom_queries), 1):
+            query = Query(custom_query)
+            query.query_data.setdefault('name', 'custom query #{}'.format(i))
+            self.queries.append(query)
+
+    def compile_queries(self):
+        transformers = TRANSFORMERS.copy()
+
+        for submission_method in SUBMISSION_METHODS:
+            method = getattr(self.check, submission_method)
+            # Save each method in the initializer -> callable format
+            transformers[submission_method] = create_submission_transformer(method)
+
+        for query in self.queries:
+            query.compile(transformers)
+
+    def execute(self):
+        logger = self.check.log
+        global_tags = self.tags
+
+        for query in self.queries:
+            query_name = query.name
+            query_columns = query.columns
+            query_tags = query.tags
+            num_columns = len(query_columns)
+
+            try:
+                rows = self.execute_query(query.query)
+            except Exception as e:
+                if self.error_handler:
+                    logger.error('Error querying %s: %s', query_name, self.error_handler(str(e)))
+                else:
+                    logger.error('Error querying %s: %s', query_name, e)
+
+                continue
+
+            for row in rows:
+                if not row:
+                    logger.debug('Query %s returned an empty result', query_name)
+                    continue
+
+                if num_columns != len(row):
+                    logger.error(
+                        'Query %s expected %d column%s, got %d',
+                        query_name,
+                        num_columns,
+                        's' if num_columns > 1 else '',
+                        len(row),
+                    )
+                    continue
+
+                row_values = {}
+                submission_queue = []
+
+                tags = list(global_tags)
+                tags.extend(query_tags)
+
+                for (column_name, transformer), value in zip(query_columns, row):
+                    # Columns can be ignored via configuration
+                    if not column_name:
+                        continue
+
+                    row_values[column_name] = value
+
+                    if callable(transformer):
+                        submission_queue.append((transformer, value))
+                    elif transformer is None:
+                        continue
+                    else:
+                        # For now there is only special logic for tags
+                        _column_type, transformer = transformer
+                        tags.append(transformer(value, None))
```

Comment:
This sections is a bit awkward. Can the transformer be None? Could we always pass a column_type?

Maybe:

```
if column_type == 'tag':
    tags.append(transformer(value, None))
else:
    submission_queue.append((transformer, value))
```

Reply:
The transformer can be `None` for `source` types. Those columns do not submit anything but are collected into the row values for other columns to reference

Reply:
I feel that: https://github.com/DataDog/integrations-core/blob/bb16139455bc5a35e04751367b03362c2322b53c/vertica/datadog_checks/vertica/vertica.py#L534..L546 is much more clear. Do you have an idea how to get closer to what we have already?